src/rollback_to_stable/rts_api.c

```diff
@@ -0,0 +1,150 @@
+/*-
+ * Copyright (c) 2014-present MongoDB, Inc.
+ * Copyright (c) 2008-2014 WiredTiger, Inc.
+ *	All rights reserved.
+ *
+ * See the file LICENSE for redistribution information.
+ */
+
+#include "wt_internal.h"
+
+/*
+ * __wt_rollback_to_stable_one --
+ *     Perform rollback to stable on a single object.
+ */
+int
+__wt_rollback_to_stable_one(WT_SESSION_IMPL *session, const char *uri, bool *skipp)
+{
+    WT_DECL_RET;
+    wt_timestamp_t rollback_timestamp;
+    char *config;
+
+    /*
+     * This is confusing: the caller's boolean argument "skip" stops the schema-worker loop from
+     * processing this object and any underlying objects it may have (for example, a table with
+     * multiple underlying file objects). We rollback-to-stable all of the file objects an object
+     * may contain, so set the caller's skip argument to true on all file objects, else set the
+     * caller's skip argument to false so our caller continues down the tree of objects.
+     */
+    *skipp = WT_BTREE_PREFIX(uri);
+    if (!*skipp)
+        return (0);
+
+    WT_RET(__wt_metadata_search(session, uri, &config));
+
+    /* Read the stable timestamp once, when we first start up. */
+    WT_ORDERED_READ(rollback_timestamp, S2C(session)->txn_global.stable_timestamp);
+
+    F_SET(session, WT_SESSION_QUIET_CORRUPT_FILE);
+    ret = __wt_rts_btree_walk_btree_apply(session, uri, config, rollback_timestamp);
+    F_CLR(session, WT_SESSION_QUIET_CORRUPT_FILE);
+
+    __wt_free(session, config);
+
+    return (ret);
+}
+
+/*
+ * __rollback_to_stable --
+ *     Rollback all modifications with timestamps more recent than the passed in timestamp.
+ */
+static int
+__rollback_to_stable(WT_SESSION_IMPL *session, bool no_ckpt)
```

Comment:
This is the implementation. Should it be in api?

Reply:
I think of this as the setup function, and __wt_rts_global_btree_apply_all as the implementation. I could see that the function could be split apart to make that more obvious (into checking, then setup and move the setup portion somewhere else).
I've tried to avoid making any code changes as part of this PR so far, so I'm reluctant to do that here.